cpmpy/expressions/variables.py

```diff
@@ -256,6 +256,9 @@ def cpm_array(arr):
     """
     if not isinstance(arr, np.ndarray):
         arr = np.array(arr)
+    elif not arr.flags['C_CONTIGUOUS'] and not arr.flags['F_CONTIGUOUS']:   # Ensure the array is contiguous
```

Comment:
Numpy actually provides a test for this: `arr.flags['FORC']`
https://numpy.org/doc/2.0/reference/generated/numpy.ndarray.flags.html